python/ray/train/v2/tests/test_state.py

```diff
@@ -415,6 +415,68 @@ def test_callback_worker_group_error(
     assert attempt.end_time_ns is not None
 
 
+def test_callback_log_file_paths(
+    ray_start_regular, monkeypatch, mock_worker_group_context, mock_worker
+):
+    """Test that StateManagerCallback correctly captures and propagates log file paths."""
+
+    # Mock the runtime context
+    mock_runtime_context = MagicMock()
+    mock_runtime_context.get_job_id.return_value = "test_job_id"
+    mock_runtime_context.get_actor_id.return_value = "test_controller_id"
+    monkeypatch.setattr(
+        ray.runtime_context, "get_runtime_context", lambda: mock_runtime_context
+    )
+
+    # Mock the log path function
+    expected_controller_log_path = (
+        "/tmp/ray/session_xxx/logs/train/ray-train-app-controller.log"
+    )
+    monkeypatch.setattr(
+        ray.train.v2._internal.callbacks.state_manager,
+        "get_train_app_controller_log_path",
+        lambda: expected_controller_log_path,
+    )
+
+    # Create the callback
+    train_run_context = TrainRunContext(RunConfig(name="test_run"))
+    callback = StateManagerCallback(train_run_context)
+
+    # Initialize the callback
+    callback.after_controller_start()
+
+    # Verify the log path was set in the state actor
+    state_actor = get_state_actor()
+    runs = ray.get(state_actor.get_train_runs.remote())
+    run = runs[callback._run_id]
+    assert run.log_file_path == expected_controller_log_path
+
+    # Now test worker log paths
+    # Create a mock worker with a log file path
+    mock_worker = mock_worker
+    mock_worker.log_file_path = (
+        "/tmp/ray/session_xxx/logs/train/ray-train-app-worker.log"
+    )
+
+    # Create a mock worker group
+    mock_worker_group = MagicMock(spec=WorkerGroup)
+    mock_worker_group.get_worker_group_context.return_value = mock_worker_group_context
+    mock_worker_group.get_worker_group_state.return_value = MagicMock(
+        workers=[mock_worker]
+    )
+    mock_worker_group.get_latest_poll_status.return_value = None
```

Comment:
Why's this needed?

Reply:
Let me double check, I think the conversion failed without it.